### PR TITLE
fix(fs): overwrite symlink in restore cache

### PR DESCRIPTION
**Pull request by AielloChan**
fix(fs): overwrite symlink in restore cache

In restore cache process, if there already exist a same link in dist folder, turbo will failed with an error:

```bash
error fetching from cache: error moving artifact from cache into XXXXX: symlink ../uuid/dist/bin/uuid XXXXX/dist/node_modules/.bin/uuid: file exists
```

This patch just remove already existed link.

### CODE REVIEW

**Review by gsoltis — CHANGES_REQUESTED** [review]
Cache restoration is currently undergoing a bunch of work (cc @nathanhammond), would you mind including a testcase with this change so that we can verify that the behavior doesn't regress?

**Comment by AielloChan** [comment]
> Cache restoration is currently undergoing a bunch of work (cc @nathanhammond), would you mind including a testcase with this change so that we can verify that the behavior doesn't regress?

Sure, I added.

File: cli/internal/fs/fs.go [hunk 1]
```diff
@@ -78,6 +78,9 @@ func CopyFile(from *LstatCachedFile, to string) error {
 		if err := EnsureDir(to); err != nil {
 			return err
 		}
+		if _, err := os.Lstat(to); err == nil {
+			os.Remove(to)
```
Comment (gsoltis) [inline cli/internal/fs/fs.go:82]: Can you handle the error here? Then I think this is ready.

Thanks!

**Review by gsoltis — CHANGES_REQUESTED** [review]

File: cli/internal/fs/fs.go (hunk 1 above)
Comment (AielloChan) [reply cli/internal/fs/fs.go]: I don't know how to properly handle `os.Lstat` error, coz it's an allowed error state. Should I handle it then write some comments?

**Review by gsoltis — APPROVED** [review]

**Pull request merged**